calicoctl/commands/node/status.go

```diff
@@ -129,21 +132,41 @@ func printBGPPeers(ipv string) {
 		os.Exit(1)
 	}
 
-	buf := make([]byte, 1024)
+	// BIRD socket read timeout.
+	timeOut := 2 * time.Second
+	bufReader := bufio.NewReader(c)
 
-	n, err := c.Read(buf[:])
-	if err != nil {
-		fmt.Printf("Error: %v\n", err)
+	birdOut := []string{}
+
+	for {
+		// Set a time-out for reading from the socket connection.
+		// Read operation will fail if no data is received until the timeout.
+		c.SetReadDeadline(time.Now().Add(timeOut))
+
+		// Read string with \n as delim.
+		str, err := bufReader.ReadString('\n')
+		if err != nil {
+			fmt.Println(err)
+			break
+		}
+
+		log.Debugf("Reading line from BIRD sock: %s\n", str)
+
+		// "0000 \n" output from BIRD means end of output.
+		if str == "0000 \n" {
+			break
+		} else {
+			// Append the line to birdOut slice of strings if it's not the end of the output.
+			birdOut = append(birdOut, str)
+		}
 	}
 
 	data := [][]string{}
 
 	table := tablewriter.NewWriter(os.Stdout)
 	table.SetHeader([]string{"Peer address", "Peer type", "State", "Since", "Info"})
 
-	birdOut := string(buf[:n])
-
-	for _, line := range strings.Split(birdOut, "\n") {
+	for _, line := range birdOut {
 
 		ipString := bgpPeerRegex.FindString(line)
 
```

Comment:
What is the separator between columns in the output data?  Is it a tab, or just some whitespace?   If it is a tab then you shouldn't use Fields(), you should split on tab instead ... since some of the info parameters might contain spaces.